lib/fog/ovirt/requests/compute/v4/add_volume.rb

```diff
@@ -0,0 +1,53 @@
+module Fog
+  module Compute
+    class Ovirt
+      class V4
+        class Real
+          DISK_SIZE_TO_GB = 1024 * 1024 * 1024
+
+          def add_volume(id, options = {})
+            raise ArgumentError, "instance id is a required parameter" unless id
+
+            options = add_options_defaults(options)
+
+            disk_attachments_service = client.system_service.vms_service.vm_service(id).disk_attachments_service
+            disk = OvirtSDK4::DiskAttachment.new(options)
+            disk_attachments_service.add(disk)
+          end
+
+          # rubocop:disable Metrics/AbcSize
+          def add_options_defaults(options)
+            search = options[:search] || format("datacenter=%<datacenter>s", :datacenter => datacenter)
+            options[:bootable] ||= "true"
+            options[:interface] ||= OvirtSDK4::DiskInterface::VIRTIO
+            options[:provisioned_size] = options[:size_gb].to_i * DISK_SIZE_TO_GB if options[:size_gb]
+
+            options[:storage_domain_id] = options[:storage_domain] || storagedomains(:role => "data", :search => search).first.id
+            # If no size is given, default to a volume size of 8GB
+            options[:provisioned_size] ||= 8 * DISK_SIZE_TO_GB
+            options[:type] ||= OvirtSDK4::DiskType::DATA
+            options[:format] ||= OvirtSDK4::DiskFormat::COW
+            options[:sparse] ||= true
```

Comment:
This way, one could never pass `:sparse => false`, as this line would always overwrite it with `true`

Reply:
We need to change that back to something like `options[:sparse] = true unless options.key?(:sparse)`. Alternatively, we could do something like

```ruby
options.reverse_merge(:sparse => true, :format => OvirtSDK4::DiskFormat::COW…)
```

Reply:
Scratch the `reverse_merge`: it's in rails, not in pure ruby